Add radio buttons for user data management options

mla04762 opened the pull request.

mla04762 committed:
Add basic radio buttons for options for managing user data

mla04762 committed:
Merge branch with main

mla04762 committed:
Add handleSubmit function for sending POST request

mla04762 committed:
Format with prettier and delete unneeded code

sharonpatta commented:
I would probably change it so that the option says "Delete User Data" instead of "Just Delete User Data"

sharonpatta committed:
Feature/vercel deployment (#15)

* Add postinstall script for vercel deployment

* Updtae postinstall deployment script

* Change server.js to access first element in data array - fixes Prisma Client Validation error (credit to chatGPT for debugging help)

* Add DATABASE_URL to example.env

* export DATABASE_URL

* revert

* Reorder steps in lint and test

* Start database in lint and test

* Change from accessing data as an array to accessing it as JSON

* Change hook for prisma generate and migrate

* Remove database startup step from github workflows

jaeminlee0515 committed:
Authenticate verification code to the database entry (#22)

* Redirect to verify page on success

* Authenticate verification code with the database entry

When the user enters in their email address, a verification code is
sent to their email. The user then input the verification code to
the redirect page. Once the verification code has been verified
with the database, it will redirect the user to a different page.
Pair programmed with Sharon and Chris.

* Fix lint errors

---------

Co-authored-by: Jae Min Lee <jaemin.lee@gordon.edu>, Sharon Patta <sharonepatta@gmail.com>

mla04762 committed:
Add function for sending POST request to Scriptoria

mla04762 committed:
Resolve confict error

mla04762 committed:
Format with prettier

mla04762 committed:
Add postUserChange function to page.server.svelte

mla04762 committed:
Move function to server.ts

mla04762 committed:
Debug lint error

chrisvire commented on tools/download-json/main.go:
```diff
@@ -33,7 +33,7 @@ type App struct {
 
 func downloadManifest(productId string, result *Manifest) error {
-	manifestUrl := fmt.Sprintf("https://app.scriptoria.io/api/products/%s/files/published/play-listing-manifest", productId)
+	manifestUrl := fmt.Sprintf("https://dev.scriptoria.io/api/products/%s/files/published/play-listing-manifest", productId)
```
Don't include this change in your commit

mla04762 replied:
I will fix this!

mla04762 committed:
Add variables to example.env

mla04762 committed:
Fix some lint formatting

mla04762 committed:
Revert change to .go file

mla04762 committed:
Verify testing with Chris and delete unneeded comment

chrisvire commented on src/routes/request/[id]/+page.server.ts:
```diff
@@ -1,10 +1,80 @@
+// src/routes/p/[id]/+page.server.ts
```
Fix comment

chrisvire commented on src/routes/request/[id]/+page.server.ts:
```diff
@@ -1,10 +1,80 @@
 
 export const load = (async ({ params: { id } }) => {
-    const request = await prisma.userManagementRequest.findUnique({
+    const app = await prisma.userManagementRequest.findUnique({
```
variable should be 'request', not 'app'

chrisvire commented on src/routes/request/[id]/+page.server.ts:
```diff
@@ -1,10 +1,80 @@
+
+export const actions = {
+    default: async ({ request, params: { id } }: RequestEvent) => {
+        console.log(SCRIPTORIA_API_TOKEN);
```
Need log?

mla04762 committed:
Change reviewed code

chrisvire commented on src/routes/request/[id]/+page.svelte:
```diff
@@ -1,9 +1,63 @@
+    import { onMount } from 'svelte';
 
-    export let data: PageData;
+    // import { API_TOKEN } from '$env/static/private';
```
Remove commented out code.

chrisvire commented on src/routes/request/[id]/+page.svelte:
```diff
@@ -1,9 +1,63 @@
+    let group = 'user datamanagement';
+    const options = { 'delete-account': 'Delete Account', 'delete-user-data': 'Delete User Data' };
+
+    onMount(() => {
```
Is this onMount function needed anymore?

If not needed, remove (and remove import)

mla04762 replied:
the onMount function is needed to make sure that there is at least one option selected.

mla04762 committed:
Change reviewed code for commented out code

mla04762 committed:
Removed maps for radio options

chrisvire committed:
Use .env SCRIPTORIA_URL for download-json

mla04762 committed:
Remove onMount() and default select delete-account

chrisvire reviewed (approved):
👍
